---
title: "Hispanic PGRS"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,dpi = 600)
knitr::opts_knit$set(root.dir = "C:/Users/timbv/Dropbox/Documents/Work")
library(knitr)
library(qqman)
library(biomaRt)
library(tidyverse)
```

# Merge and Quality Control

All sample quality control steps were performed using plink version 2.0 (http://www.cog-genomics.org/plink/2.0/) unless otherwise specified, and all merge steps were performed using plink version 1.9 (http://www.cog-genomics.org/plink/1.9/). Duplicate variants and indels were deleted from each dataset separately prior to merging, because the Biobank dataset included allele information from indels, whereas the Simmons data did not. After these steps, there were 515,855 variants remaining in the Simmons dataset and 1,844,663 in the Biobank dataset (the discrepancy in the total number of variants is due to the fact that the Simmons data had already passed some quality control steps for preliminary analyses).

A first attempt to merge the datasets revealed 231,289 variants with inconsistent alleles between the two datasets. After strand-flipping these variants, there remained 1,125 genomic positions with inconsistent variants. Variant IDs, position, and alleles were manually checked using the programming language R version 4.0.2 (https://www.r-project.org/). 524 variants appeared to be genuinely different between the two datasets, and all other variants were the same but with listed positions off by a single base pair. The 524 different variants were excluded from further analyses, while the remaining variants were re-named in the Biobank dataset to match the Simmons data naming conventions.

After resolving the naming and position discrepancies, the two datasets were successfully merged and the unified dataset underwent sample quality control. Quality control measures included a call rate (< 0.02), discrepancy between reported and genotype-inferred sex, minor allele frequency (MAF) > 0.95, Hardy-Weinberg equilibrium in controls (P > $10^{-10}$), and sample heterozygosity. Samples were also checked for relatedness using the KING-robust kinship coefficient, although no individuals were excluded due to relatedness. After quality control measures, 100,365 SNPs in 667 cases and 968 controls remained.

## Quality Control Steps

1. 15 people excluded due to missing genotype data (723 remaining after this)
2. 36 participants excluded due to sec check discrepancies (687 remaining)
3. 20 participants excluded due to heterozygosity rate (667 remaining). Heterozygosity rate is the number of SNPs for which an individual has two different alleles. High heterozygosity rate suggests that the sample was low quality.

```{r}
# Get SNPs of interest
snps_interest <- read.csv("./simmons/Variants in any T1D GRS.csv",
                          stringsAsFactors = F)
# Get positions from Ensembl
snp_mart = useMart(biomart="ENSEMBL_MART_SNP", host="grch37.ensembl.org",
                   dataset="hsapiens_snp")
bm <- getBM(attributes = c('refsnp_id','chr_name',
                           'chrom_start','chrom_end','allele'), 
            filters = c("snp_filter"),
            values = snps_interest$SNP,mart = snp_mart)
bm$combined <- paste0(bm$chr_name,":",bm$chrom_start)
# Check
simmons <- read.delim("./Simmons_DataDeliverable_061919/Simmons_MEGA1_Deliverable_06142019/cleaned_files/Simmons_Custom_MEGA_Analysi_03012019_snpFailsRemoved_passing_QC.bim",header = F,stringsAsFactors = F)
simmons$combined <- paste0(simmons$V1,":",simmons$V4)
simmons_snps <- bm$refsnp_id[bm$combined %in% simmons$combined]

biobank <- read.delim("./simmons/Simmons_071520.bim",stringsAsFactors = F,header = F)
biobank$combined <- paste0(biobank$V1,":",biobank$V4)
biobank_snps <- bm$refsnp_id[bm$combined %in% biobank$combined]
# See how many Illumina chip (the one used in TEDDY, etc.) are available
illumina <- read.csv("./simmons/illumina_snps.csv",header = F,
                     stringsAsFactors = F)
illumina_pos <- 
  getBM(attributes = c('refsnp_id','chr_name',
                       'chrom_start','chrom_end','allele'), 
        filters = c("snp_filter"),
        values = illumina[,1],mart = snp_mart)
illumina_pos$combined <- 
  paste0(illumina_pos$chr_name,":",illumina_pos$chrom_start)
```

```{r indels}
simmons <- read.delim("./Simmons_passed_qc.bim",header = F)
write.table(simmons$V2[which(simmons$V5 == "I" | simmons$V5 == "D")],
            file = "./qc/exclude_indels",row.names = F,
            quote = F,col.names = F)
```

```{bash eval=FALSE}
# Change directory
cd /Users/timvigers/Documents/simmons/
# Remove indels and duplicates
plink --bfile Simmons_passed_qc --exclude qc/exclude_indels --make-bed --out qc/simmons
plink --bfile qc/simmons --list-duplicate-vars ids-only suppress-first --out qc/simmons
plink --bfile qc/simmons --exclude qc/simmons.dupvar --make-bed --out qc/simmons

plink --bfile Simmons_071520 --snps-only --make-bed --out qc/biobank
plink --bfile qc/biobank --list-duplicate-vars ids-only suppress-first --out qc/biobank
plink --bfile qc/biobank --exclude qc/biobank.dupvar --make-bed --out qc/biobank
# Merge files
plink --bfile qc/simmons --bmerge qc/biobank --make-bed --out qc/first_merge
# This produces multiple position warnings, so try flipping
plink --bfile qc/simmons --flip qc/first_merge-merge.missnp --make-bed --out qc/simmons_flipped
# Re-merge flipped
plink --bfile qc/simmons_flipped --bmerge qc/biobank --make-bed --out qc/flip_merge
```

```{r name differences}
# Import .bim files
simmons <- read.delim("./qc/simmons.bim", header=FALSE)
biobank <- read.delim("./qc/biobank.bim", header=FALSE)
# Check that the matching positions have the same alleles (or can be flipped to be the same)
simmons$position <- paste(simmons$V1,simmons$V4,sep = "_")
biobank$position <- paste(biobank$V1,biobank$V4,sep = "_")
matching <- intersect(simmons$position,biobank$position)
# Combine
simmons <- simmons[simmons$position %in% matching,]
biobank <- biobank[biobank$position %in% matching,]
all <- full_join(simmons,biobank,by="position")
# Check names
all <- all[which(all$V2.x != all$V2.y),]
# Check alleles
ex <- all[which(all$V5.x != all$V5.y),]
# Check flipped
ex <- ex[which(ex$V5.x != ex$V6.y),]
# Remove variants that are genuinely different
ex <- c(as.character(ex$V2.x),as.character(ex$V2.y))
# Change names for all others
all <- all[!(all$V2.x %in% ex),]
all <- all[!(all$V2.y %in% ex),]
# Write for exclusion
write.table(ex,file = "./qc/exclude_mismatch",row.names = F,
            quote = F,col.names = F)
# Change names in R
biobankbim <- read.delim("./qc/biobank.bim", header=FALSE)
biobankbim$V2[match(all$V2.y,biobankbim$V2)] <- all$V2.x
write.table(biobankbim,"./qc/biobank.bim",row.names = F,quote = F,col.names = F)
```

```{bash eval=FALSE}
# Exclude name mismatches and SNPs that are different
plink --bfile qc/biobank --make-bed --out qc/biobank --exclude qc/exclude_mismatch
plink --bfile qc/biobank --make-bed --out qc/biobank --exclude qc/first_merge-merge.missnp
plink --bfile qc/simmons --make-bed --out qc/simmons --exclude qc/exclude_mismatch
plink --bfile qc/simmons --make-bed --out qc/simmons --exclude qc/first_merge-merge.missnp
# Merge again
plink --bfile qc/simmons --bmerge qc/biobank --make-bed --out qc/biobank_simmons_merge
# QC merged data
plink2 --bfile qc/biobank_simmons_merge --geno 0.02 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --mind 0.02 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --maf 0.05 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --mac 20 --make-bed --out qc/biobank_simmons_merge # for GLM
plink2 --bfile qc/biobank_simmons_merge --hwe 1e-6 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --hwe 1e-10 --make-bed --out qc/biobank_simmons_merge
# Prune
plink2 --bfile qc/biobank_simmons_merge --indep-pairwise 50 5 0.2 --out qc/indepSNP
plink2 --bfile qc/biobank_simmons_merge --extract qc/indepSNP.prune.in --make-bed --out analysis/biobank_simmons_merge_final
# Check kinship - none (good!)
plink2 --bfile analysis/biobank_simmons_merge_final --make-king-table --king-table-filter 0.354 --out qc/biobank_simmons_kin
```

```{r ensembl,cache=TRUE}
# Check how many of the SNPs of interest are in the final merge
merged_snps <- read.delim("./qc/biobank_simmons_merge.bim",
                          header=FALSE)
# Ensembl
snp_mart = useMart(biomart="ENSEMBL_MART_SNP", host="grch37.ensembl.org",
                   dataset="hsapiens_snp")
bm <- getBM(attributes = c('refsnp_id','chr_name',
                           'chrom_start','chrom_end','allele'),
            filters = c("snp_filter"),
            values = Variants.in.any.T1D.GRS$SNP,mart = snp_mart)
# Check
bm$combined <- paste0(bm$chr_name,":",bm$chrom_start)
merged_snps$combined <- paste0(merged_snps$V1,":",merged_snps$V4)
```

# PCA

```{bash eval=FALSE}
plink2 --bfile analysis/biobank_simmons_merge_final --pca --out analysis/PCA
```

## By Cohort

There are clear batch effects between Kimber's data and the Biobank data. When PC3 is included in the logistic regression, the VIF for cohort is really high, suggesting that PC3 is primarily driven by batch.

```{r pca by batch}
# Clear workspace
rm(list = ls())
# PCA
PCA <- read.delim("./analysis/PCA.eigenvec")
colnames(PCA)[1] <- "FID"
# Get ethnicity
kimber <- read.csv("./clinical.csv")
biobank <- read.csv("./Biobank Hispanic Patients - Compass Mapping File.csv")
PCA$studyid <- gsub(".*_","",PCA$IID)
PCA$Ethnicity <- kimber$Hispanic..0.N..1.Y.[match(PCA$studyid,kimber$correct_labID)]
PCA$Ethnicity[is.na(match(PCA$studyid,kimber$correct_labID))] <- 1
PCA$Ethnicity <- factor(PCA$Ethnicity,levels = c(0,1),labels = c("Non-Hispanic","Hispanic"))
# Cohort info
PCA$Cohort <- ifelse(PCA$studyid %in% kimber$correct_labID,"Kimber","Biobank")
# Export Kimber only
write.table(PCA[PCA$Cohort == "Kimber",1:2],"./analysis/kimber_ids",
            row.names = F,col.names = F,quote = F)
# Sex
PCA$Sex <- kimber$Gender[match(PCA$studyid,kimber$correct_labID)]
PCA$Sex[is.na(PCA$Sex)] <- biobank$Sex[match(PCA$studyid[is.na(PCA$Sex)],biobank$Arb_person_id)]
PCA$Sex <- as.factor(PCA$Sex)
levels(PCA$Sex) <- c("F", "F", "M", "M")
# Plot (colorblind friendly)
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ggplot(PCA,aes(x=PC1,y=PC2)) +
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC2,y=PC3)) +
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC1,y=PC3)) +
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
```

```{r}
# Make covariate txt files
write.table(PCA[,c("FID","IID","Sex","PC1","PC2")],
            file = "./analysis/covar.txt",quote = F,row.names = F)
write.table(PCA[,c("FID","IID","Sex","PC1","PC2","PC3")],
            file = "./analysis/covar_pc3.txt",quote = F,row.names = F)
```

# Analysis

```{bash eval=FALSE}
# Logistic regression
plink2 --bfile analysis/biobank_simmons_merge_final --covar analysis/covar.txt  --logistic 'firth-fallback' --out analysis/logistic_results
plink2 --bfile analysis/biobank_simmons_merge_final --covar analysis/covar_pc3.txt  --logistic 'firth-fallback' --out analysis/logistic_results_pc3
# Adjusted p values
plink2 --adjust-file analysis/logistic_results.PHENO1.glm.logistic.hybrid 'test=ADD' --out analysis/logistic_results
plink2 --adjust-file analysis/logistic_results_pc3.PHENO1.glm.logistic.hybrid 'test=ADD' --out analysis/logistic_results_pc3
```

# Results without adjusting for PC3 (batch effect)

```{r logistic results,warning=FALSE}
# Import plink results
logistic_results <- read.delim("./analysis/logistic_results.PHENO1.glm.logistic.hybrid")
# Format data
res <- logistic_results[logistic_results$TEST == "ADD",c("ID","X.CHROM","POS","P","Z_STAT")]
colnames(res) <- c("SNP","CHR","BP","P")
# Get SNPs of interest from Kimber
snps_interest <- read.csv("./Variants in any T1D GRS.csv")
# Plot
manhattan(res,main = "Manhattan Plot",highlight = snps_interest$SNP)
qq(res$P)
```

# Adjusted results

```{r logistic pc3 results,warning=FALSE}
# Import plink results
logistic_results <- read.delim("./analysis/logistic_results_pc3.PHENO1.glm.logistic.hybrid")
# Format data
res <- logistic_results[logistic_results$TEST == "ADD",c("ID","X.CHROM","POS","P","Z_STAT")]
colnames(res) <- c("SNP","CHR","BP","P")
# Get SNPs of interest from Kimber
snps_interest <- read.csv("./Variants in any T1D GRS.csv")
# Plot
manhattan(res,main = "Manhattan Plot",highlight = snps_interest$SNP)
qq(res$P)
```

# HLA imputation

```{bash eval=FALSE}
csh -f ~/Documents/Work/GWAS/SNP2HLA_package_v1.0.3/SNP2HLA/SNP2HLA.csh analysis/biobank_simmons_merge_final ~/Documents/Work/GWAS/SNP2HLA_package_v1.0.3/SNP2HLA/HM_CEU_REF analysis/biobank_simmons_merge_final_imputed plink
```

```{bash eval=FALSE}
# Delete temp files
find . -type f -name "*~" -delete
```
