---
title: "Hispanic PGRS"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = "/Users/timvigers/Documents/OneDrive - The University of Colorado Denver/simmons")
library(arsenal)
library(skimr)
library(knitr)
library(tidyverse)
```

```{r indels}
simmons <- read.delim("./Simmons_passed_qc.bim",header = F)
write.table(simmons$V2[which(simmons$V5 == "I" | simmons$V5 == "D")],
            file = "./qc/exclude_indels",row.names = F,
            quote = F,col.names = F)
```

```{bash}
# Change directory
cd /Users/timvigers/Documents/OneDrive\ -\ The\ University\ of\ Colorado\ Denver/simmons/
# Remove indels and duplicates
plink --bfile Simmons_passed_qc --exclude qc/exclude_indels --make-bed --out qc/simmons
plink --bfile qc/simmons --list-duplicate-vars ids-only suppress-first --out qc/simmons
plink --bfile qc/simmons --exclude qc/simmons.dupvar --make-bed --out qc/simmons

plink --bfile Simmons_071520 --snps-only --make-bed --out qc/biobank
plink --bfile qc/biobank --list-duplicate-vars ids-only suppress-first --out qc/biobank
plink --bfile qc/biobank --exclude qc/biobank.dupvar --make-bed --out qc/biobank
# Merge files
plink --bfile qc/simmons --bmerge qc/biobank --make-bed --out qc/first_merge
# This produces multiple position warnings, so try flipping 
plink --bfile qc/simmons --flip qc/first_merge-merge.missnp --make-bed --out qc/simmons_flipped
# Re-merge flipped
plink --bfile qc/simmons_flipped --bmerge qc/biobank --make-bed --out qc/flip_merge
```

```{r name differences}
# Import .bim files
simmons <- read.delim("./qc/simmons.bim", header=FALSE)
biobank <- read.delim("./qc/biobank.bim", header=FALSE)
# Check that the matching positions have the same alleles (or can be flipped to be the same)
simmons$position <- paste(simmons$V1,simmons$V4,sep = "_")
biobank$position <- paste(biobank$V1,biobank$V4,sep = "_")
matching <- intersect(simmons$position,biobank$position)
# Combine
simmons <- simmons[simmons$position %in% matching,]
biobank <- biobank[biobank$position %in% matching,]
all <- full_join(simmons,biobank,by="position")
# Check names
all <- all[which(all$V2.x != all$V2.y),]
# Check alleles
ex <- all[which(all$V5.x != all$V5.y),]
# Check flipped
ex <- ex[which(ex$V5.x != ex$V6.y),]
# Remove variants that are genuinely different
ex <- c(as.character(ex$V2.x),as.character(ex$V2.y))
# Change names for all others
all <- all[!(all$V2.x %in% ex),]
all <- all[!(all$V2.y %in% ex),]
# Write for exclusion
write.table(ex,file = "./qc/exclude_mismatch",row.names = F,
            quote = F,col.names = F)
# Change names in R
biobankbim <- read.delim("./qc/biobank.bim", header=FALSE)
biobankbim$V2[match(all$V2.y,biobankbim$V2)] <- all$V2.x
write.table(biobankbim,"./qc/biobank.bim",row.names = F,quote = F,col.names = F)
```

```{bash}
# Exclude name mismatches and SNPs that are different 
plink --bfile qc/biobank --make-bed --out qc/biobank --exclude qc/exclude_mismatch
plink --bfile qc/biobank --make-bed --out qc/biobank --exclude qc/first_merge-merge.missnp
plink --bfile qc/simmons --make-bed --out qc/simmons --exclude qc/exclude_mismatch
plink --bfile qc/simmons --make-bed --out qc/simmons --exclude qc/first_merge-merge.missnp
# Merge again
plink --bfile qc/simmons --bmerge qc/biobank --make-bed --out qc/biobank_simmons_merge
# QC merged data
plink2 --bfile qc/biobank_simmons_merge --geno 0.02 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --mind 0.02 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --maf 0.05 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --mac 20 --make-bed --out qc/biobank_simmons_merge # for GLM
plink2 --bfile qc/biobank_simmons_merge --hwe 1e-6 --make-bed --out qc/biobank_simmons_merge
plink2 --bfile qc/biobank_simmons_merge --hwe 1e-10 --make-bed --out qc/biobank_simmons_merge
# Prune
plink2 --bfile qc/biobank_simmons_merge --indep-pairwise 50 5 0.2 --out qc/indepSNP
plink2 --bfile qc/biobank_simmons_merge --extract qc/indepSNP.prune.in --make-bed --out analysis/biobank_simmons_merge_final
# Check kinship - none (good!)
plink2 --bfile analysis/biobank_simmons_merge_final --make-king-table --king-table-filter 0.354 --out qc/biobank_simmons_kin
```

# PCA 

```{bash}
plink2 --bfile analysis/biobank_simmons_merge_final --pca --out analysis/PCA
```

## By Cohort

There are clear batch effects between Kimber's data and the Biobank data. When PC3 is included in the logistic regression, the VIF for cohort is really high, suggesting that PC3 is primarily driven by batch. 

```{r pca by batch}
# Clear workspace
rm(list = ls())
# PCA
PCA <- read.delim("./analysis/PCA.eigenvec")
colnames(PCA)[1] <- "FID"
# Get ethnicity
kimber <- read.csv("./clinical.csv")
biobank <- read.csv("./Biobank Hispanic Patients - Compass Mapping File.csv")
PCA$studyid <- gsub(".*_","",PCA$IID)
PCA$Ethnicity <- kimber$Hispanic..0.N..1.Y.[match(PCA$studyid,kimber$correct_labID)]
PCA$Ethnicity[is.na(match(PCA$studyid,kimber$correct_labID))] <- 1
PCA$Ethnicity <- factor(PCA$Ethnicity,levels = c(0,1),labels = c("Non-Hispanic","Hispanic"))
# Cohort info
PCA$Cohort <- ifelse(PCA$studyid %in% kimber$correct_labID,"Kimber","Biobank")
# Export Kimber only
write.table(PCA[PCA$Cohort == "Kimber",1:2],"./analysis/kimber_ids",
            row.names = F,col.names = F,quote = F)
# Sex
PCA$Sex <- kimber$Gender[match(PCA$studyid,kimber$correct_labID)]
PCA$Sex[is.na(PCA$Sex)] <- biobank$Sex[match(PCA$studyid[is.na(PCA$Sex)],biobank$Arb_person_id)]
PCA$Sex <- as.factor(PCA$Sex)
levels(PCA$Sex) <- c("F", "F", "M", "M")
# Plot (colorblind friendly)
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ggplot(PCA,aes(x=PC1,y=PC2)) + 
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC2,y=PC3)) + 
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC1,y=PC3)) + 
  geom_point(aes(color = Cohort)) + scale_color_manual(values=cbPalette) + theme_bw()
```

## By Ethnicity

Question for Nick/Laura: It doesn't look like Hispanic vs. non-Hispanic are separating well, does that mean that the first two PCs aren't really representing population structure? Would it be better to project these samples onto the 1000 Genomes data and then adjust for PCs?

```{r pca by ethnicity}
ggplot(PCA,aes(x=PC1,y=PC2)) + 
  geom_point(aes(color = Ethnicity)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC2,y=PC3)) + 
  geom_point(aes(color = Ethnicity)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(PCA,aes(x=PC1,y=PC3)) + 
  geom_point(aes(color = Ethnicity)) + scale_color_manual(values=cbPalette) + theme_bw()
```

```{r}
# Make covariate txt file
write.table(PCA[,c("FID","IID","Sex","PC1","PC2","PC3")],
            file = "./analysis/covar.txt",quote = F,row.names = F)
```

# Analysis

```{bash}
# Logistic regression
plink2 --bfile analysis/biobank_simmons_merge_final --keep analysis/kimber_ids --covar analysis/covar.txt  --logistic 'firth-fallback' --out analysis/logistic_results
# Adjusted p values
plink2 --adjust-file analysis/logistic_results.PHENO1.glm.logistic.hybrid 'test=ADD' --out analysis/logistic_results
```

```{r}
# Import plink results
logistic_results <- read.delim("./analysis/logistic_results.PHENO1.glm.logistic.hybrid")
adjusted_results <- read.delim("./analysis/logistic_results.adjusted")
```

```{bash}
# Delete temp files
find . -type f -name "*~" -delete
```

