---
title: "Hispanic T1D MEGA Array QC"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = "C:/Users/timbv/Desktop/QC")
library(tidyverse)
library(qqman)
```

# Quality Control

All sample quality control steps were performed using plink version 2.0 (http://www.cog-genomics.org/plink/2.0/) unless otherwise specified, and all merge steps were performed using plink version 1.9 (http://www.cog-genomics.org/plink/1.9/). Duplicate variants and indels were deleted, after which there were 1,943,170 variants remaining in the dataset.

Quality control measures included a call rate (< 0.02), discrepancy between reported and genotype-inferred sex, minor allele frequency (MAF) > 0.95, Hardy-Weinberg equilibrium (P > $10^{-10}$), and sample heterozygosity. Samples were also checked for relatedness using the KING-robust kinship coefficient, although no individuals were excluded due to relatedness. Phase 3 data from the 1000 Genomes project (Auton et al., 2015) were used to examine population stratification and confirm Hispanic heritage.

After quality control measures, 135,626 SNPs in 667 participants remained.

## Quality Control Steps

1. 15 people excluded due to missing genotype data (723 remaining after this)
2. 36 participants excluded due to sex check discrepancies (687 remaining)
3. 20 participants excluded due to heterozygosity rate (667 remaining). Heterozygosity rate is the number of SNPs for which an individual has two different alleles. High heterozygosity rate suggests that the sample was low quality.

```{bash combine files,eval=FALSE,results = 'hide'}
# Combine files
plink --bfile Simmons_redo --bmerge Simmons_removeKnownFails --make-bed --out Simmons_combined
```

```{r check combine,echo=FALSE,eval=FALSE}
combined <- read.table("Simmons_combined.fam")
combined$V6 <- 2
write.table(combined, file = "./Simmons_combined.fam",row.names = F,
            sep = "\t",col.names = F,quote = F)
clinical <- read.csv("./clinical.csv")
remove <- combined[which(!(sub(".*_","",combined$V2) %in% clinical$correct_labID)),1:2]
write.table(remove, file = "./NAs_remove.txt",row.names = F,sep = "\t",col.names = F,quote = F)
```

### Missingness pre-deletion

```{bash plink missing,eval=FALSE,results = 'hide'}
# Remove IDs with NA
plink --bfile Simmons_combined --remove NAs_remove.txt --make-bed --out Simmons_combined
# Check missing
plink --bfile Simmons_combined --missing
```

```{r r missing plots pre-deletion,echo=FALSE}
# Read in plink missingness files
indmiss <-read.table(file="./plink.imiss",
                     header=TRUE)
snpmiss <-read.table(file="./plink.lmiss",
                     header=TRUE)
# Plots 
hist(indmiss[,6],main="Histogram proportion missing SNPs per individual",
     xlab = "Proportion Missing")
hist(snpmiss[,5],main="Histogram proportion missing individuals per SNP",
     xlab = "Proportion Missing ") 
```

### Missingness post-deletion

```{bash plink filter missing,eval=FALSE,results = 'hide'}
# Delete SNPs and individuals with high levels of missingness
# Delete SNPs
plink --bfile Simmons_combined --geno 0.02 --make-bed --out Simmons_combined_1
# Delete individuals
plink --bfile Simmons_combined_1 --mind 0.02 --make-bed --out Simmons_combined_1
# Check missing post-deletion
plink --bfile Simmons_combined_1 --missing --out miss_post_del
```

```{r r missing plots post-deletion,echo=FALSE}
# Post-deletion
indmiss_check <-
  read.table(file="./miss_post_del.imiss",
             header=TRUE)
snpmiss_check <-
  read.table(file="./miss_post_del.lmiss",
             header=TRUE)
# Plots
hist(indmiss_check[,6],main="Histogram proportion missing SNPs per individual",
     xlab = "Proportion Missing")
hist(snpmiss_check[,5],main="Histogram proportion missing individuals per SNP",
     xlab = "Proportion Missing ") 
# For duplicated individuals, keep the one with highest call rate (lowest missing)
indmiss_check$id <- gsub(".*_","",indmiss_check$IID)
keep <- indmiss_check %>% group_by(id) %>% slice(which.min(N_MISS))
write.table(keep[,1:2],file = "./call_rate_keep.txt",row.names = F,sep = "\t",col.names = F,quote = F)
```

```{bash plink call rate,eval=FALSE,results = 'hide'}
# Delete low call rate duplicates
plink --bfile Simmons_combined_1 --keep call_rate_keep.txt --make-bed --out Simmons_combined_2
```

```{r echo=FALSE}
# Read in deleted 
post_repeated_subs_del <- read.table("./Simmons_combined_2.fam") 
```

### Sex check pre-deletion

```{bash plink sex check pre,eval=FALSE,results = 'hide'}
# Check sex discrepancy
plink --bfile Simmons_combined_2 --check-sex
```

```{r r sex check pre,echo=FALSE}
# Gender file
gender <- read.table("./plink.sexcheck",
                     header=T,as.is=T)
# Fix known incorrectly labelled sex
correct_sex <-read.csv("./clinical.csv")
correct_sex$Gender <- ifelse(correct_sex$Gender == "M",1,2)
sex_fam <- read.table("./Simmons_combined_2.fam")
sex_fam$V5 <- 
  correct_sex$Gender[match(as.character(sub(".*_","",sex_fam$V2)),as.character(correct_sex$correct_labID))]
# Overwrite
write.table(sex_fam,file = "./Simmons_combined_2.fam",
            row.names = F,col.names = F,quote = F)
# Histograms
hist(gender[,6],main="Gender", xlab="F")
# Males
male=subset(gender, gender$PEDSEX==1)
hist(male[,6],main="Men",xlab="F")
# Females
female=subset(gender, gender$PEDSEX==2)
hist(female[,6],main="Women",xlab="F")
```

### Sex check post-deletion

```{bash plink sex check post,eval=FALSE,results = 'hide'}
# Remove subjects with problematic X chromosome homozygosity
grep "PROBLEM" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt
plink --bfile Simmons_combined_2 --remove sex_discrepancy.txt --make-bed --out Simmons_combined_3
# Check sex discrepancy again
plink --bfile Simmons_combined_3 --check-sex --out sex_post_del
```

```{r r sex check post, echo=FALSE}
# Gender file
gender <- read.table("./sex_post_del.sexcheck",
                     header=T,as.is=T)
# Histograms
hist(gender[,6],main="Gender", xlab="F")
# Males
male=subset(gender, gender$PEDSEX==1)
hist(male[,6],main="Men",xlab="F")
# Females
female=subset(gender, gender$PEDSEX==2)
hist(female[,6],main="Women",xlab="F")
```

### MAF check pre-deletion

```{bash plink maf check pre,eval=FALSE,results = 'hide'} 
# Select autosomal SNPs only (i.e., from chromosomes 1 to 22).
awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' Simmons_combined_3.bim > snp_1_22.txt
plink --bfile Simmons_combined_3 --extract snp_1_22.txt --make-bed --out Simmons_combined_4
# Check minor allele frequency
plink --bfile Simmons_combined_4 --freq --out MAF_check
```

```{r r maf check pre, echo=FALSE}
# MAF file
maf_freq <- read.table("./MAF_check.frq", header =TRUE, as.is=T)
hist(maf_freq[,5],main = "MAF distribution", xlab = "MAF")
```

### MAF check post-deletion

```{bash plink maf check post,eval=FALSE,results = 'hide'}
# Remove those with MAF < 0.05
plink --bfile Simmons_combined_4 --maf 0.05 --make-bed --out Simmons_combined_5
# Check minor allele frequency again
plink --bfile Simmons_combined_5 --freq --out MAF_check_post
```

```{r r maf check post, echo=FALSE}
# MAF file
maf_freq <- read.table("./MAF_check_post.frq", header =TRUE, as.is=T)
hist(maf_freq[,5],main = "MAF distribution", xlab = "MAF")
```

### Hardy-Weinberg equilibrium pre-deletion

```{bash plink hw check pre,eval=FALSE,results = 'hide'}
plink --bfile Simmons_combined_5 --hardy
# Select SNPs with HWE p-value below 0.00001
awk '{ if ($9 <0.00001) print $0 }' plink.hwe>plinkzoomhwe.hwe
```

```{r r hw check pre, echo=FALSE}
hwe<-read.table (file="./plink.hwe", header=TRUE)
hist(hwe[,9],main="Histogram HWE")
# Zoomed in
hwe_zoom<-read.table (file="./plinkzoomhwe.hwe", header=TRUE)
hist(hwe_zoom[,9],main="Histogram HWE: strongly deviating SNPs only")
```

### Hardy-Weinberg equilibrium post-deletion

```{bash plink hw check post,eval=FALSE,results = 'hide'}
# Filter for controls
plink --bfile Simmons_combined_5 --hwe 1e-6 --make-bed --out Simmons_combined_6
# Filter for cases
plink --bfile Simmons_combined_6 --hwe 1e-12 --hwe-all --make-bed --out Simmons_combined_6
# Check again
plink --bfile Simmons_combined_6 --hardy --out post_del
# Select SNPs with HWE p-value below 0.00001
awk '{ if ($9 <0.00001) print $0 }' post_del.hwe>post_del_zoom.hwe
```

```{r r hw check post, echo=FALSE}
hwe<-read.table (file="./post_del.hwe", header=TRUE)
hist(hwe[,9],main="Histogram HWE")
# Zoomed in
hwe_zoom<-read.table (file="./post_del_zoom.hwe", header=TRUE)
hist(hwe_zoom[,9],main="Histogram HWE: strongly deviating SNPs only")
```

### Heterozygosity pre-deletion

```{bash plink heterozyg check pre,eval=FALSE,results = 'hide'}
plink --bfile Simmons_combined_6 --indep-pairwise 50 5 0.2 --out indepSNP
plink --bfile Simmons_combined_6 --extract indepSNP.prune.in --het --out R_check
```

```{r r heterozyg check pre, echo=FALSE}
het <- read.table("./R_check.het", head=TRUE)
het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM."
hist(het$HET_RATE, xlab="Heterozygosity Rate", ylab="Frequency", main= "Heterozygosity Rate")
# List individuals more than 3 SD from the rate mean
het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM."
het_fail = subset(het, (het$HET_RATE < mean(het$HET_RATE)-3*sd(het$HET_RATE)) | (het$HET_RATE > mean(het$HET_RATE)+3*sd(het$HET_RATE)));
het_fail$HET_DST = (het_fail$HET_RATE-mean(het$HET_RATE))/sd(het$HET_RATE);
write.table(het_fail, "fail-het-qc.txt", row.names=FALSE)
```

### Heterozygosity post-deletion

```{bash plink heterozyg check post,eval=FALSE,results = 'hide'}
sed 's/"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt
# Remove heterozygosity rate outliers.
plink --bfile Simmons_combined_6 --remove het_fail_ind.txt --make-bed --out Simmons_passed_qc
plink --bfile Simmons_passed_qc --extract indepSNP.prune.in --het --out R_check_post
```

```{r r heterozyg check post, echo=FALSE}
het <- read.table("./R_check_post.het", head=TRUE)
het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM."
hist(het$HET_RATE, xlab="Heterozygosity Rate", ylab="Frequency", main= "Heterozygosity Rate")
```

```{bash plink related check pre,eval=FALSE,results = 'hide'}
plink2 --bfile Simmons_passed_qc --make-king-table --king-table-filter 0.354 --out Simmons_passed_qc_kin
```

# Population Stratification

```{bash make TGP file,eval=FALSE}
# Remove variants based on missing genotype data, write to Kimber's folder.
plink2 --bfile all_phase3_qc --geno 0.2 --allow-no-sex --make-bed --out 1kG_MDS --allow-extra-chr
```

```{bash TGP QC,eval=FALSE}
# Remove individuals based on missing genotype data.
plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2 --allow-extra-chr
# Remove variants based on MAF.
plink --bfile 1kG_MDS2 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS3 --allow-extra-chr
# Exclude all non-autosomal variants, except those with chromosome code XY. Filter out founders.
plink --bfile 1kG_MDS3 --allow-extra-chr --allow-no-sex --autosome-xy --make-bed --out 1kG_MDS4 --filter-founders
# TGP data QC done
# Allele frequency
plink --bfile Simmons_passed_qc --freq --out Simmons_passed_qc
```

```{r exclude at and cg,echo=FALSE,eval=FALSE}
# Find common SNPs
kimber = read.delim("./Simmons_passed_qc.bim",header=F, quote="")
tgp = read.delim("./1kG_MDS4.bim", header=F, quote="")
common.snps = which(tgp$V2 %in% kimber$V2)
common.snps <- as.character(tgp$V2[common.snps])

alleles = read.table("./Simmons_passed_qc.frq", header=T)
alleles$A <- paste0(alleles$A1,alleles$A2)
alleles <- alleles %>% filter(!(A %in% c("AT","TA","GC","CG")))
snps <- common.snps[common.snps %in% alleles$SNP]
write.table(snps,file="./list.snps",sep="\t",col.names=F,
            row.names=F,quote=F)
```

```{bash plink exclude at and cg,eval=FALSE}
# Extract the variants present in Kimber's dataset from the 1000 genomes dataset.
plink --bfile 1kG_MDS4 --extract list.snps --make-bed --out 1kG_MDS5
# Vice versa
plink --bfile Simmons_passed_qc --extract list.snps --make-bed --out Simmons_MDS
```

```{bash merge prune mds,eval=FALSE}
# Make a list of multiple-position SNPs
plink --bfile Simmons_MDS --bmerge 1kG_MDS5.bed 1kG_MDS5.bim 1kG_MDS5.fam --make-bed --out first_merge
# Exclude
plink --bfile Simmons_MDS --make-bed --out Simmons_trimmed --exclude first_merge-merge.missnp
plink --bfile 1kG_MDS5 --make-bed --out 1kG_MDS6 --exclude first_merge-merge.missnp
# Merge again
plink --bfile Simmons_trimmed --bmerge 1kG_MDS6.bed 1kG_MDS6.bim 1kG_MDS6.fam --make-bed --out MDS_merge
# Prune
plink --bfile MDS_merge --indep-pairwise 50 5 0.2 --out mergeSNP
plink --bfile MDS_merge --extract mergeSNP.prune.in --make-bed --out final_merge
# PCA
plink --bfile final_merge --cluster --pca --out PCA
# Convert MDS and PCA to covariate files
awk '{print$1, $2, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13}' final_merge_mds.mds > covar_mds.txt
```

### PCA

```{r echo=FALSE,warning=FALSE}
# Read in race info for Kimber's participants
kimber <- read.csv("./clinical.csv")
# Match to .fam IDs
fam <- read.table(file="./Simmons_MDS.fam",header=F)
fam$studyid <- gsub(".*_","",fam$V2)
fam$race <- 
  kimber$Hispanic..0.N..1.Y.[match(fam$studyid,kimber$correct_labID)]
fam$race <- 
  factor(fam$race,levels = c(0,1),labels = c("Non-Hispanic","Hispanic"))
fam$race <- as.character(fam$race)
fam$Gender <- 
  kimber$Gender[match(fam$studyid,kimber$correct_labID)]
# Reference races
race<- read.csv(file="./race_1kG.csv",stringsAsFactors = F)
colnames(race) <- c("FID","V2","superpop","pop")
race <- full_join(race,fam[,c("V2","race","Gender")],by = "V2")
race <- race %>% mutate(pop = coalesce(race,pop))
race$superpop <- 
  ifelse(race$pop == "Hispanic","Hispanic",
         ifelse(race$pop == "Non-Hispanic","Non-Hispanic",race$superpop))
colnames(race)[2] <- "IID"

data<- read.table(file="./PCA.eigenvec",header=FALSE)
colnames(data) <- c("FID","IID",paste0("C",seq(1,ncol(data)-2)))
datafile<- left_join(data,race[,c("IID","superpop","pop","Gender")],
                     by="IID")
# Add sex
ref_sex <- read.csv("./G1K_sample_info.csv")
datafile$Gender[datafile$IID %in% ref_sex$IID] <- 
  ref_sex$Gender[match(datafile$IID,ref_sex$IID)]
datafile$Gender <- as.factor(datafile$Gender)
levels(datafile$Gender) <- c("F","F","M","M")
# Plot (colorblind friendly)
cbPalette <- c("#999999","#E69F00","#56B4E9","#009E73","#F0E442",
               "#0072B2","#D55E00","#CC79A7")
ggplot(datafile,aes(x=C1,y=C2)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(datafile,aes(x=C2,y=C3)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(datafile,aes(x=C1,y=C3)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
# Make covariate txt files
write.table(datafile[,c("FID","IID","Gender","C1","C2","C3")],
            file = "./covar.txt",quote = F,row.names = F)
```

```{r scree,echo=FALSE}
scree <- read.table("./PCA.eigenval")
colnames(scree) <- "var"
scree$percent_explained <- round((scree$var / sum(scree$var))*100,3)
scree$pc <- paste0("PC",seq(1,nrow(scree)))
scree$pc <- factor(scree$pc,levels = scree$pc)
ggplot(scree,aes(x = factor(pc),y = percent_explained)) + geom_point() + 
  xlab("") + ylab("% Variation Explained") + ggtitle("PCA Scree Plot") + theme_bw()
```

# Analysis

```{bash eval=FALSE}
# Just Kimber's data
plink2 --bfile simmons_passed_qc --indep-pairwise 50 5 0.2 --out simmons_mergeSNP
plink2 --bfile simmons_passed_qc --extract simmons_mergeSNP.prune.in --make-bed --out simmons_pruned
# PCA
plink2 --bfile simmons_pruned --pca --out PCA
```

```{r}
# Change race to "phenotype"
simmons <- read.table("./simmons_pruned.fam")
clinical <- read.csv("./clinical.csv")
simmons$V6 <- 
  clinical$Hispanic..0.N..1.Y.[match(gsub(".*_","",simmons$V2),
                                     clinical$correct_labID)] + 1
write.table(simmons, file = "./simmons_pruned.fam",row.names = F,
            sep = "\t",col.names = F,quote = F)
# Make covariate text file
covar <- read.table("./pca.eigenvec")
covar$id <- gsub(".*_","",covar$V2)
covar$sex <- clinical$Gender[match(covar$id,clinical$correct_labID)]
covar <- covar[,c("V1","V2","sex","V3","V4")]
colnames(covar) <- c("FID","IID","sex","PC1","PC2")
# Write
write.table(covar,file = "./covar.txt",quote = F,row.names = F)
```

```{bash eval=FALSE}
# Logistic regression
plink2 --bfile simmons_pruned --covar covar.txt --logistic --out logistic_results
# Adjusted p values
plink2 --adjust-file logistic_results.PHENO1.glm.logistic.hybrid 'test=ADD' --out logistic_results
```

# Results

```{r logistic results,warning=FALSE}
# Import plink results
logistic_results <- read.delim("./logistic_results.PHENO1.glm.logistic.hybrid")
# Format data
res <- logistic_results[logistic_results$TEST == "ADD",c("ID","X.CHROM","POS","P","Z_STAT")]
colnames(res) <- c("SNP","CHR","BP","P","Z")
# Get SNPs of interest from Kimber
snps_interest <- read.csv("./Variants in any T1D GRS.csv")
# Plot
manhattan(res,main = "Manhattan Plot")
qq(res$P)
# Write significant SNPs
write.csv(res[res$P<0.05,],file = "./significant_snps.csv",row.names = F)
```
