---
title: "GWAS"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = "/Users/timvigers/Dropbox/Work/Kimber Simmons/GWAS")
library(snpStats)
```

# Quality Control

```{r import and merge,cache=TRUE}
# Kimber
path <- paste("/Users/timvigers/Dropbox/Work/Kimber Simmons/GWAS/Data_Raw/Simmons_MEGA1_Deliverable_06142019/cleaned_files/Simmons_Custom_MEGA_Analysi_03012019_snpFailsRemoved_passing_QC", c(".bed", ".bim", ".fam"), sep = "")
redo <- read.plink(path[1], path[2], path[3])
# First biobank
path <- paste("/Users/timvigers/Dropbox/Work/Kimber Simmons/GWAS/Data_Raw/Simmons Biobank/Simmons_071520", c(".bed", ".bim", ".fam"), sep = "")
biobank1 <- read.plink(path[1], path[2], path[3])
# Second biobank
path <- paste("/Users/timvigers/Dropbox/Work/Kimber Simmons/GWAS/Data_Raw/V2 - Biobank data on Hispanic Patients - Full Genetic Request/Simmons_120420", c(".bed", ".bim", ".fam"), sep = "")
biobank2 <- read.plink(path[1], path[2], path[3])
# Merge
SNP <- rbind(redo$genotypes, biobank1$genotypes, biobank2$genotypes)
```