---
title: "GWAS"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/home/tim/Work/Kimber Simmons/GWAS/Data_Cleaned/plink")
library(tidyverse)
```

# QC

## Missingness pre-deletion

```{r r missing plots pre-deletion,echo=FALSE}
# Read in plink missingness files
indmiss <-read.table(file="./plink2.smiss",
                    header=TRUE)
snpmiss <-read.table(file="./plink2.vmiss",
                    header=TRUE)
# Plots 
hist(indmiss[,6],main="Histogram proportion missing SNPs per individual",
     xlab = "Proportion Missing")
hist(snpmiss[,5],main="Histogram proportion missing individuals per SNP",
     xlab = "Proportion Missing ") 
```

## Missingness post-deletion

```{r r missing plots post-deletion,echo=FALSE}
# Post-deletion
indmiss_check <- read.table(file="./miss_post_del.smiss",header=TRUE)
snpmiss_check <- read.table(file="./miss_post_del.vmiss",header=TRUE)
# Plots
hist(indmiss_check[,6],main="Histogram proportion missing SNPs per individual",
     xlab = "Proportion Missing")
hist(snpmiss_check[,5],main="Histogram proportion missing individuals per SNP",
     xlab = "Proportion Missing ") 
```

### Sex check pre-deletion

```{r r sex check pre, echo=FALSE}
# Gender file
gender <- read.table("./plink.sexcheck",header=T,as.is=T)
# Histograms
hist(gender[,6],main="Gender", xlab="F")
# Males
male=subset(gender, gender$PEDSEX==1)
hist(male[,6],main="Men",xlab="F")
# Females
female=subset(gender, gender$PEDSEX==2)
hist(female[,6],main="Women",xlab="F")
```

<!-- ### Sex check post-deletion -->

<!-- ```{bash plink sex check post,eval=FALSE,results = 'hide'} -->
<!-- # Remove subjects with problematic X chromosome homozygosity -->
<!-- grep "PROBLEM" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt -->
<!-- plink --bfile Simmons_combined_2 --remove sex_discrepancy.txt --make-bed --out Simmons_combined_3 -->
<!-- # Check sex discrepancy again -->
<!-- plink --bfile Simmons_combined_3 --check-sex --out sex_post_del -->
<!-- ``` -->

<!-- ```{r r sex check post, echo=FALSE} -->
<!-- # Gender file -->
<!-- gender <- read.table("./sex_post_del.sexcheck", -->
<!--                      header=T,as.is=T) -->
<!-- # Histograms -->
<!-- hist(gender[,6],main="Gender", xlab="F") -->
<!-- # Males -->
<!-- male=subset(gender, gender$PEDSEX==1) -->
<!-- hist(male[,6],main="Men",xlab="F") -->
<!-- # Females -->
<!-- female=subset(gender, gender$PEDSEX==2) -->
<!-- hist(female[,6],main="Women",xlab="F") -->
<!-- ``` -->

<!-- ### MAF check pre-deletion -->

<!-- ```{bash plink maf check pre,eval=FALSE,results = 'hide'}  -->
<!-- # Select autosomal SNPs only (i.e., from chromosomes 1 to 22). -->
<!-- awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' Simmons_combined_3.bim > snp_1_22.txt -->
<!-- plink --bfile Simmons_combined_3 --extract snp_1_22.txt --make-bed --out Simmons_combined_4 -->
<!-- # Check minor allele frequency -->
<!-- plink --bfile Simmons_combined_4 --freq --out MAF_check -->
<!-- ``` -->

<!-- ```{r r maf check pre, echo=FALSE} -->
<!-- # MAF file -->
<!-- maf_freq <- read.table("./MAF_check.frq", header =TRUE, as.is=T) -->
<!-- hist(maf_freq[,5],main = "MAF distribution", xlab = "MAF") -->
<!-- ``` -->

<!-- ### MAF check post-deletion -->

<!-- ```{bash plink maf check post,eval=FALSE,results = 'hide'} -->
<!-- # Remove those with MAF < 0.05 -->
<!-- plink --bfile Simmons_combined_4 --maf 0.05 --make-bed --out Simmons_combined_5 -->
<!-- # Check minor allele frequency again -->
<!-- plink --bfile Simmons_combined_5 --freq --out MAF_check_post -->
<!-- ``` -->

<!-- ```{r r maf check post, echo=FALSE} -->
<!-- # MAF file -->
<!-- maf_freq <- read.table("./MAF_check_post.frq", header =TRUE, as.is=T) -->
<!-- hist(maf_freq[,5],main = "MAF distribution", xlab = "MAF") -->
<!-- ``` -->

<!-- ### Hardy-Weinberg equilibrium pre-deletion -->

<!-- ```{bash plink hw check pre,eval=FALSE,results = 'hide'} -->
<!-- plink --bfile Simmons_combined_5 --hardy -->
<!-- # Select SNPs with HWE p-value below 0.00001 -->
<!-- awk '{ if ($9 <0.00001) print $0 }' plink.hwe>plinkzoomhwe.hwe -->
<!-- ``` -->

<!-- ```{r r hw check pre, echo=FALSE} -->
<!-- hwe<-read.table (file="./plink.hwe", header=TRUE) -->
<!-- hist(hwe[,9],main="Histogram HWE") -->
<!-- # Zoomed in -->
<!-- hwe_zoom<-read.table (file="./plinkzoomhwe.hwe", header=TRUE) -->
<!-- hist(hwe_zoom[,9],main="Histogram HWE: strongly deviating SNPs only") -->
<!-- ``` -->

<!-- ### Hardy-Weinberg equilibrium post-deletion -->

<!-- ```{bash plink hw check post,eval=FALSE,results = 'hide'} -->
<!-- # Filter for controls -->
<!-- plink --bfile Simmons_combined_5 --hwe 1e-6 --make-bed --out Simmons_combined_6 -->
<!-- # Filter for cases -->
<!-- plink --bfile Simmons_combined_6 --hwe 1e-12 --hwe-all --make-bed --out Simmons_combined_6 -->
<!-- # Check again -->
<!-- plink --bfile Simmons_combined_6 --hardy --out post_del -->
<!-- # Select SNPs with HWE p-value below 0.00001 -->
<!-- awk '{ if ($9 <0.00001) print $0 }' post_del.hwe>post_del_zoom.hwe -->
<!-- ``` -->

<!-- ```{r r hw check post, echo=FALSE} -->
<!-- hwe<-read.table (file="./post_del.hwe", header=TRUE) -->
<!-- hist(hwe[,9],main="Histogram HWE") -->
<!-- # Zoomed in -->
<!-- hwe_zoom<-read.table (file="./post_del_zoom.hwe", header=TRUE) -->
<!-- hist(hwe_zoom[,9],main="Histogram HWE: strongly deviating SNPs only") -->
<!-- ``` -->

<!-- ### Heterozygosity pre-deletion -->

<!-- ```{bash plink heterozyg check pre,eval=FALSE,results = 'hide'} -->
<!-- plink --bfile Simmons_combined_6 --exclude inversion.txt --range --indep-pairwise 50 5 0.2 --out indepSNP -->
<!-- plink --bfile Simmons_combined_6 --extract indepSNP.prune.in --het --out R_check -->
<!-- ``` -->

<!-- ```{r r heterozyg check pre, echo=FALSE} -->
<!-- het <- read.table("./R_check.het", head=TRUE) -->
<!-- het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM." -->
<!-- hist(het$HET_RATE, xlab="Heterozygosity Rate", ylab="Frequency", main= "Heterozygosity Rate") -->
<!-- # List individuals more than 3 SD from the rate mean -->
<!-- het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM." -->
<!-- het_fail = subset(het, (het$HET_RATE < mean(het$HET_RATE)-3*sd(het$HET_RATE)) | (het$HET_RATE > mean(het$HET_RATE)+3*sd(het$HET_RATE))); -->
<!-- het_fail$HET_DST = (het_fail$HET_RATE-mean(het$HET_RATE))/sd(het$HET_RATE); -->
<!-- write.table(het_fail, "fail-het-qc.txt", row.names=FALSE) -->
<!-- ``` -->

<!-- ### Heterozygosity post-deletion -->

<!-- ```{bash plink heterozyg check post,eval=FALSE,results = 'hide'} -->
<!-- sed 's/"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt -->
<!-- # Remove heterozygosity rate outliers. -->
<!-- plink --bfile Simmons_combined_6 --remove het_fail_ind.txt --make-bed --out Simmons_passed_qc -->
<!-- plink --bfile Simmons_passed_qc --extract indepSNP.prune.in --het --out R_check_post -->
<!-- ``` -->

<!-- ```{r r heterozyg check post, echo=FALSE} -->
<!-- het <- read.table("./R_check_post.het", head=TRUE) -->
<!-- het$HET_RATE = (het$"N.NM." - het$"O.HOM.")/het$"N.NM." -->
<!-- hist(het$HET_RATE, xlab="Heterozygosity Rate", ylab="Frequency", main= "Heterozygosity Rate") -->
<!-- ``` -->

<!-- ### Relatedness -->

<!-- ```{bash plink related check pre,eval=FALSE,results = 'hide'} -->
<!-- plink --bfile Simmons_passed_qc --genome --min 0.05 -->
<!-- ``` -->

<!-- ```{r r related check pre, echo=FALSE} -->
<!-- relatedness = read.table("./plink.genome", header=T) -->
<!-- par(pch=16, cex=1) -->
<!-- with(relatedness,plot(Z0,Z1, xlim=c(0,1), ylim=c(0,1), type="n")) -->
<!-- with(subset(relatedness,RT=="UN") , points(Z0,Z1,col=3)) -->
<!-- legend(1,1, xjust=1, yjust=1, legend=levels(relatedness$RT), pch=16, col=c(3)) -->
<!-- hist(relatedness[,10],main="Histogram relatedness", xlab= "Pihat")  -->
<!-- ``` -->