---
title: "MEGA Population Stratification"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(rgl)
library(tidyverse)
```

### QC TGP Data

```{bash TGP QC,eval=FALSE}
cd /Volumes/som/PEDS/RI\ Biostatistics\ Core/Shared
# QC reference
# Remove variants based on missing genotype data, write to Kimber's folder.
plink --bfile Shared\ Material/Reference\ Genomes/plink_TGP/all_phase3_ns --geno 0.2 --allow-no-sex --make-bed --out Shared\ Projects/Laura/BDC/Projects/Kimber\ Simmons/GWAS/Data_Cleaned/population_stratification/1kG_MDS --allow-extra-chr --memory 6144
# Move to population stratification folder
cd Shared\ Projects/Laura/BDC/Projects/Kimber\ Simmons/GWAS/Data_Cleaned/population_stratification
# Remove individuals based on missing genotype data.
plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2 --allow-extra-chr --memory 6144
# Remove variants based on MAF.
plink --bfile 1kG_MDS2 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS3 --allow-extra-chr --memory 6144
# Exclude all non-autosomal variants, except those with chromosome code XY. Filter out founders.
plink --bfile 1kG_MDS3 --allow-extra-chr --allow-no-sex --autosome-xy --make-bed --out 1kG_MDS4 --memory 6144 --filter-founders
# TGP data QC done
# Allele frequency
plink --bfile Simmons_passed_qc --freq --memory 6144 --out Simmons_passed_qc
```

```{r exclude at and cg,echo=FALSE,eval=FALSE}
# Find common SNPs
kimber = read.delim("/Volumes/Tim/kimber/population_stratification/Simmons_passed_qc.bim", 
                    header=F, quote="")
tgp = read.delim("/Volumes/Tim/kimber/population_stratification/1kG_MDS4.bim", header=F, quote="")
common.snps = which(tgp$V2 %in% kimber$V2)
common.snps <- as.character(tgp$V2[common.snps])

alleles = read.table("/Volumes/Tim/kimber/population_stratification/Simmons_passed_qc.frq",
                     header=T)
alleles$A <- paste0(alleles$A1,alleles$A2)
alleles <- alleles %>% filter(!(A %in% c("AT","TA","GC","CG")))
snps <- common.snps[common.snps %in% alleles$SNP]
write.table(snps,file="/Volumes/Tim/kimber/population_stratification/list.snps", 
            sep="\t",col.names=F,row.names=F,quote=F)
```

```{bash plink exclude at and cg,eval=FALSE}
# Extract the variants present in Kimber's dataset from the 1000 genomes dataset.
plink --bfile 1kG_MDS4 --extract list.snps --make-bed --out 1kG_MDS5 --memory 6144
# Vice versa
plink --bfile Simmons_passed_qc --extract list.snps --make-bed --out Simmons_MDS --memory 6144
```

### Merge

```{bash merge prune mds,eval=FALSE}
# Make a list of multiple-position SNPs
plink --bfile Simmons_MDS --bmerge 1kG_MDS5.bed 1kG_MDS5.bim 1kG_MDS5.fam --make-bed --out first_merge --memory 6144
# Exclude
plink --bfile Simmons_MDS --make-bed --out Simmons_trimmed --memory 6144 --exclude first_merge-merge.missnp
plink --bfile 1kG_MDS5 --make-bed --out 1kG_MDS6 --memory 6144 --exclude first_merge-merge.missnp
# Merge again
plink --bfile Simmons_trimmed --bmerge 1kG_MDS6.bed 1kG_MDS6.bim 1kG_MDS6.fam --make-bed --out MDS_merge --memory 6144
# Prune
plink --bfile MDS_merge --indep-pairwise 50 5 0.2 --out mergeSNP --memory 6144
plink --bfile MDS_merge --extract mergeSNP.prune.in --make-bed --out final_merge --memory 6144
# PCA
plink --bfile final_merge --cluster --pca --out PCA --memory 6144
```

### PCA

```{r echo=FALSE,warning=FALSE}
# Read in race info for Kimber's participants
kimber <- read.csv("/Volumes/Tim/kimber/population_stratification/clinical.csv")
# Match to .fam IDs
fam <- read.table(file="/Volumes/Tim/kimber/population_stratification/Simmons_MDS.fam",header=F)
fam$studyid <- gsub(".*_","",fam$V2)
fam$race <- kimber$Hispanic..0.N..1.Y.[match(fam$studyid,kimber$correct_labID)]
fam$race <- factor(fam$race,levels = c(0,1),labels = c("Non-Hispanic","Hispanic"))
fam$race <- as.character(fam$race)
# Reference races
race<- read.csv(file="/Volumes/Tim/kimber/population_stratification/race_1kG.csv",
                stringsAsFactors = F)
colnames(race) <- c("FID","V2","superpop","pop")
race <- full_join(race,fam[,c("V2","race")],by = "V2")
race <- race %>% mutate(pop = coalesce(race,pop))
race$superpop <- ifelse(race$pop == "Hispanic","Hispanic",
                        ifelse(race$pop == "Non-Hispanic","Non-Hispanic",race$superpop))
colnames(race)[2] <- "IID"

data<- read.table(file="/Volumes/Tim/kimber/population_stratification/PCA.eigenvec",
                  header=FALSE)
colnames(data) <- c("FID","IID",paste0("pc",seq(1,ncol(data)-2)))

datafile<- left_join(data,race[,c("IID","superpop","pop")],by="IID")
# Plot (colorblind friendly)
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ggplot(datafile,aes(x=pc1,y=pc2)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(datafile,aes(x=pc2,y=pc3)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
ggplot(datafile,aes(x=pc1,y=pc3)) + geom_point(aes(color = superpop)) + scale_color_manual(values=cbPalette) + theme_bw()
```

```{r 3d plot,echo=FALSE,eval=FALSE}
plot3d(datafile[,c("pc1","pc2","pc3")],col = as.integer(factor(datafile$superpop)))
```

```{r scree,echo=FALSE}
scree <- read.table("/Volumes/Tim/kimber/population_stratification/PCA.eigenval")
colnames(scree) <- "var"
scree$percent_explained <- round((scree$var / sum(scree$var))*100,3)
scree$pc <- paste0("PC",seq(1,nrow(scree)))
scree$pc <- factor(scree$pc,levels = scree$pc)
ggplot(scree,aes(x = factor(pc),y = percent_explained)) + geom_point() + 
  xlab("") + ylab("% Variation Explained") + ggtitle("PCA Scree Plot") + theme_bw()
```