---
title: "MEGA Population Stratification"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Volumes/Tim/Kimber_GWAS/Data_Cleaned/Population_Stratification")
library(tidyverse)
```



```{bash make 1000 genomes bed bim fam,cache=TRUE,results = 'hide'}
# Convert 1000 genomes file to plink format
plink --vcf ALL.2of4intersection.20100804.genotypes.vcf.gz --make-bed --out ALL.2of4intersection.20100804.genotypes
# Set missing IDs
plink --bfile ALL.2of4intersection.20100804.genotypes --set-missing-var-ids @:#[b37]\$1,\$2 --make-bed --out ALL.2of4intersection.20100804.genotypes_no_missing_IDs
```

```{bash 1000 genomes QC,cache=TRUE,results = 'hide'}
# Remove variants based on missing genotype data.
plink --bfile ALL.2of4intersection.20100804.genotypes_no_missing_IDs --geno 0.2 --allow-no-sex --make-bed --out 1kG_MDS
# Remove individuals based on missing genotype data.
plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2
# Remove variants based on missing genotype data.
plink --bfile 1kG_MDS2 --geno 0.02 --allow-no-sex --make-bed --out 1kG_MDS3
# Remove individuals based on missing genotype data.
plink --bfile 1kG_MDS3 --mind 0.02 --allow-no-sex --make-bed --out 1kG_MDS4
# Remove variants based on MAF.
plink --bfile 1kG_MDS4 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS5
```

```{bash extract variants,cache=TRUE,results = 'hide'}
# Simmons from 1000 genomes
awk '{print$2}' Simmons_passed_qc.bim > Simmons_SNPs.txt
plink --bfile 1kG_MDS5 --extract Simmons_SNPs.txt --make-bed --out 1kG_MDS6
# 1000 genomes from Simmons
awk '{print$2}' 1kG_MDS6.bim > 1kG_MDS6_SNPs.txt
plink --bfile Simmons_passed_qc --extract 1kG_MDS6_SNPs.txt --recode --make-bed --out Simmons_MDS
```