---
title: "MEGA Population Stratification"
author: "Tim Vigers & Laura Pyle"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(tidyverse)
```

### Extract TGP Data

```{bash plink2,eval=FALSE}
cd /Volumes/Tim/
# Decompress .zst file
plink2 --zst-decompress plink_TGP/all_phase3_ns.pgen.zst plink_TGP/all_phase3_ns.pgen
# Convert plink2 TGP files to plink1
plink2 --pgen plink_TGP/all_phase3_ns.pgen --pvar plink_TGP/all_phase3_ns.pvar.zst --psam plink_TGP/phase3_corrected.psam --maf 0.05 --make-bed --out plink_TGP/all_phase3_ns --max-alleles 2 --memory 6144
```

### QC TGP Data

```{bash eval=FALSE}
# QC reference
# Remove variants based on missing genotype data, write to Kimber's folder.
plink --bfile plink_TGP/all_phase3_ns --geno 0.2 --allow-no-sex --make-bed --out kimber/population_stratification/1kG_MDS --allow-extra-chr --memory 6144
cd /Volumes/Tim/kimber/population_stratification/
# Remove individuals based on missing genotype data.
plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2 --allow-extra-chr --memory 6144
# Remove variants based on MAF.
plink --bfile 1kG_MDS2 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS3 --allow-extra-chr --memory 6144
# Exclude all non-autosomal variants, except those with chromosome code XY. Filter out founders.
plink --bfile 1kG_MDS3 --allow-extra-chr --allow-no-sex --autosome-xy --make-bed --out 1kG_MDS4 --memory 6144 --filter-founders
# TGP data QC done
```

```{r same variants,echo=FALSE,eval=FALSE}
# Find common SNPs
kimber = read.delim("/Volumes/Tim/kimber/population_stratification/Simmons_passed_qc.bim", header=F, quote="")
tgp = read.delim("/Volumes/Tim/kimber/population_stratification/1kG_MDS4.bim", header=F, quote="")
common.snps = which(tgp$V2 %in% kimber$V2)
write.table(tgp$V2[common.snps],file="/Volumes/Tim/kimber/population_stratification/list.snps", 
            sep="\t",col.names=F,row.names=F,quote=F)
```

### Merge

```{bash merge prune mds,eval=FALSE}
# Extract the variants present in Kimber's dataset from the 1000 genomes dataset.
plink --bfile 1kG_MDS4 --extract list.snps --make-bed --out 1kG_MDS5
# Vice versa
plink --bfile Simmons_passed_qc --extract list.snps --make-bed --out Simmons_MDS
# Make a list of multiple-position SNPs
plink --bfile Simmons_MDS --bmerge 1kG_MDS5.bed 1kG_MDS5.bim 1kG_MDS5.fam --make-bed --out first_merge --memory 6144
# Exclude
plink --bfile Simmons_MDS --make-bed --out Simmons_trimmed --memory 6144 --exclude first_merge-merge.missnp
plink --bfile 1kG_MDS5 --make-bed --out 1kG_MDS6 --memory 6144 --exclude first_merge-merge.missnp
# Merge again
plink --bfile Simmons_trimmed --bmerge 1kG_MDS6.bed 1kG_MDS6.bim 1kG_MDS6.fam --make-bed --out MDS_merge --memory 6144
# Prune
plink --bfile MDS_merge --indep-pairwise 50 5 0.2 --out mergeSNP --memory 6144
plink --bfile MDS_merge --extract mergeSNP.prune.in --make-bed --out final_merge --memory 6144
# PCA
plink --bfile final_merge --cluster --pca --out PCA --memory 6144
```

### PCA

```{r echo=FALSE,warning=FALSE}
# Read in race info for Kimber's participants
kimber <- read.csv("/Volumes/Tim/kimber/population_stratification/clinical.csv")
# Match to .fam IDs
fam <- read.table(file="/Volumes/Tim/kimber/population_stratification/test.fam",header=F)
fam$studyid <- gsub(".*_","",fam$V2)
fam$race <- kimber$Hispanic..0.N..1.Y.[match(fam$studyid,kimber$correct_labID)]
fam$race <- factor(fam$race,levels = c(0,1),labels = c("Non-Hispanic","Hispanic"))
fam$race <- as.character(fam$race)
# Reference races
race<- read.csv(file="/Volumes/Tim/kimber/population_stratification/race_1kG.csv",
                stringsAsFactors = F)
colnames(race) <- c("FID","V2","superpop","pop")
race <- full_join(race,fam[,c("V2","race")],by = "V2")
race <- race %>% mutate(pop = coalesce(race,pop))
race$superpop <- ifelse(race$pop == "Hispanic","Hispanic",
                        ifelse(race$pop == "Non-Hispanic","Non-Hispanic",race$superpop))
colnames(race)[2] <- "IID"

data<- read.table(file="/Volumes/Tim/kimber/population_stratification/PCA.eigenvec",
                  header=FALSE)
colnames(data) <- c("FID","IID",paste0("pc",seq(1,ncol(data)-2)))

datafile<- left_join(data,race[,c("IID","superpop","pop")],by="IID")
# Plot
ggplot(datafile,aes(x=pc1,y=pc2)) + geom_point(aes(color = superpop)) + theme_bw()
ggplot(datafile,aes(x=pc2,y=pc3)) + geom_point(aes(color = superpop)) + theme_bw()
ggplot(datafile,aes(x=pc1,y=pc3)) + geom_point(aes(color = superpop)) + theme_bw()
```

### Association

```{r change pheno to study}
study_pheno <- read.table("/Volumes/Tim/kimber/population_stratification/final_merge.fam")
study_pheno$V6[study_pheno$V6 == -9] <- 1 # TGP are "controls"
write.table(study_pheno,file="/Volumes/Tim/kimber/population_stratification/study_pheno.fam", 
            sep="\t",col.names=F,row.names=F,quote=F)
```

```{bash regression,eval=FALSE}
plink --bfile --bim final_merge.bim --bed final_merge.bed --fam study_pheno.fam --logistic --out logistic_results
```

```{r logistic results}
log_results <- read.table("/Volumes/Tim/kimber/population_stratification/logistic_results.assoc.logistic",
                          header = T)
log_results <- log_results %>% arrange(P)
kable(log_results[1:50,])
```